### PR TITLE
Add backwards-compatible API for LSRK4.

### PR DESCRIPTION
**Pull request by MTCam**
Function rename for LSRK4 caused outstanding drivers using it to fail.  This patch brings back a backwards-compatible interface with a warning. cc @thomasgibson @anderson2981

### CODE REVIEW

**Comment by anderson2981** [comment]
I'm not sure if we need this. At least for now we live in a rapidly evolving state of software development. I think it might be best to just update the drivers with the correct syntax and move forward. See also, the checkpointing refactor.

I think maybe it just confuses things down the road.

**Comment by MTCam** [comment]
> I'm not sure if we need this. At least for now we live in a rapidly evolving state of software development. I think it might be best to just update the drivers with the correct syntax and move forward. (See also #257)

As a practical matter, I could not agree more, and would pursue this approach myself.  However, we need to use software process to ensure that we don't break driver-facing APIs in the future.  Dealing with immediate destructive domino effects of API changes is something that can be 100% avoided if we make sure to maintain backwards-compatibility.

This also applies to #257, but this situation there is going to be a bit tougher.

> I think maybe it just confuses things down the road.

I'm concerned with avoiding the confusing thing in the road right in front of me (i.e. Y1 Nozzle monitoring is currently failing due to driver-facing API changes), and trying to avoid it happening again.

**Comment by anderson2981** [comment]
> 
> I'm concerned with avoiding the confusing thing in the road right in front of me (i.e. Y1 Nozzle monitoring is currently failing due to driver-facing API changes), and trying to avoid it happening again.

I think you're off the hook with regards to performance. At least for now, that driver is not compatible with master. There is an approved version of mirgecom saved with the driver. TBH, this will probably be updated today and become a non-issue. I just haven't gotten to it yet. 

How to approach this might make a good topic for our Friday meeting.

**Comment by MTCam** [comment]
Maybe @inducer will be a more friendly audience for this PR.

File: mirgecom/integrators/lsrk.py [hunk 1]
```diff
@@ -34,6 +34,14 @@
 import numpy as np
 
 
+def lsrk4_step(state, t, dt, rhs):
+    """Call lsrk54_step with backwards-compatible interface."""
+    from warnings import warn
+    warn("Do not call lsrk4; it is now callled lsrk54_step. This function will "
+         "disappear April 1, 2021", DeprecationWarning, stacklevel=2)
```
Comment (inducer) [inline mirgecom/integrators/lsrk.py:41]: A month is probably a bit short. I usually do at least a few months.

**Review by inducer — APPROVED** [review]
I'm the one who suggested this, and in a way, this prompted some thinking on the mirgecom -> driver interface stability, for which this minor thing just provides the occasion. My thought process behind this is that (if not now, then at least some way down the road) we may commit to a defined "stable API surface" that mirgecom exhibits towards drivers. On occasion, this may necessitate compatibility shims like what's in this PR, for example here to restore the driver-facing API that was broken in #265 . I generally view these compatibility shims as low-cost, and I don't mind the little bit of clutter that they add. To be clear, sometimes interface changes are necessary that are too big to be bridged by this type of thing. But minor refactors are easily covered like this. FWIW, roughly this is the policy I already follow with many of the other packages.

cc @thomasgibson

**Comment by inducer** [comment]
Btw, I agree with @anderson2981  that this makes  a good discussion point for the Friday meeting.

File: mirgecom/integrators/lsrk.py [hunk 2]
```diff
@@ -34,6 +34,14 @@
 import numpy as np
 
 
+def lsrk4_step(state, t, dt, rhs):
```
[the remark this answers is not in the document]
Comment (MTCam) [reply mirgecom/integrators/lsrk.py:37]: Sure. `integrators` is now a directory, though. Any hints on how to handle that?  Ignore it?

[the remark this answers is not in the document]
Comment (majosm) [reply mirgecom/integrators/lsrk.py:37]: I think you'd just have to import it into `integrators/__init__.py`, right?

[the remark this answers is not in the document]
Comment (MTCam) [reply mirgecom/integrators/lsrk.py:37]: :shrug: Seems like a good guess.

File: mirgecom/integrators/lsrk.py (hunk 1 above)
Comment (MTCam) [reply mirgecom/integrators/lsrk.py]: (fbe4769)

File: mirgecom/integrators/lsrk.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (MTCam) [reply mirgecom/integrators/lsrk.py]: (fbe4769)

**Comment by MTCam** [comment]
Tested that this works in fixing @anderson2981's Y1 Nozzle drivers.

**Pull request merged**